app/src/navigation/screens/DocumentDetailScreen.tsx

```diff
@@ -0,0 +1,209 @@
+import * as React from "react";
+import { View, StyleSheet, Text } from "react-native";
+import { connectActionSheet } from "@expo/react-native-action-sheet";
+import { Button, FileList, Modal } from "../../components";
+import { useDocumentStore } from "../../context/DocumentStore";
+import {
+  DocumentsStackNavigation,
+  DocumentsStackNavigationRoute
+} from "../../types/navigation";
+import { useNavigation, useRoute } from "@react-navigation/native";
+import BottomNavBarSpacer from "../../components/BottomNavBarSpacer";
+import { ScrollView } from "react-native-gesture-handler";
+import { Input } from "@rneui/themed";
+import useSelectPhoto from "../../hooks/useSelectPhoto";
+import { DOCUMENT_IMAGE_OPTIONS } from "../../utils/image-utils";
+import * as DocumentPicker from "expo-document-picker";
+
+type Navigation = DocumentsStackNavigation<"DocumentDetail">;
+
+type FormState = {
+  [key: string]: string;
+};
+
+const DocumentDetailScreen: React.FC = () => {
+  const navigation = useNavigation<Navigation>();
+  const route = useRoute<DocumentsStackNavigationRoute<"DocumentDetail">>();
+  const document = route.params.document;
+
+  const { files, addFile, deleteFile, addDocument } = useDocumentStore();
+  const { selectPhotoFromCameraRoll, selectPhotoFromCamera } = useSelectPhoto(
+    DOCUMENT_IMAGE_OPTIONS
+  );
+
+  const [showAttachModal, setShowAttachModal] = React.useState<boolean>(false);
+  const [formState, setFormState] = React.useState<FormState>({});
+  const [updateDocs, setUpdateDocs] = React.useState<boolean>(false);
+
+  const selectedDocuments = React.useMemo(
+    () => files.filter((file) => document.fileIds?.includes(file.id)),
+    [files, updateDocs]
+  );
+
+  React.useEffect(() => {
+    document.fileIds = document.fileIds ? document.fileIds : [];
+    const newForm = { ...formState };
+    document.fields?.forEach((field) => {
+      newForm[field.title] = field.value ?? "";
+    });
+    setFormState(newForm);
+  }, []);
+
+  const attachFile = () => {
+    setShowAttachModal(true);
```

Comment:
probably not worth creating a new function just to call this

Reply:
i think it's good separation of concern

Reply:
Not sure if the code guidelines cover it or not, just seems wrong to me to define a function that performs no calculations, doesn't return anything, and only calls another function. If it is a valid separation of concerns then feel free to resolve again.

Reply:
We were adviced a while ago that on react native its faster to avoid declaring any function inline and that as a best practice all functions should be defined at the top and only called inline.  Admittedly we haven't stuck very strongly to this but doing it this way is best practice.